src/components/LibrariesPage.tsx

```diff
@@ -76,14 +71,64 @@ export class LibrariesPage extends React.Component<LibrariesPageProps, Libraries
     this.props.fetchData();
   }
 
+  filter(libraries) {
+    return libraries.filter(l => l.stages.registry_stage === "production");
+  }
+
+  getLibraryList() {
+    // Does the registry have libraries?
+    let hasLibraries = (this.props.libraries && this.props.libraries.libraries && this.props.libraries.libraries.length);
+    // List of all libraries for the registry:
+    let allLibraries =  hasLibraries && this.props.libraries.libraries;
+    // There might be QA libraries available, but the user only wants to see the production ones:
+    let filteredLibraries = !this.state.qa && allLibraries && this.filter(allLibraries);
+    // The user has submitted a search, and there are results:
+    let allResults = !this.state.showAll && this.props.results && this.props.results.libraries;
+    // The user has submitted a search, but only wants to see results for production libraries:
+    let filteredResults = !this.state.qa && allResults && this.filter(allResults);
+    // The user has submitted a search, but there are no results:
+    let noResults = !this.state.showAll && !this.props.results && [];
+
+    // There are six possible scenarios:
+    // 1) A search has been submitted, QA mode is active, and there are search results. Set the libraries variable to the list of results.
+    // 2) A search has been submitted, QA mode is not active, and there are search results.  Set the libraries variable to a filtered list
+    // of results--just the ones that are in production.
+    // 3) A search has been submitted, and there are not search results. Set the libraries variable to an empty array;
+    // this will generate a "no results were found" message.
+    // 4) A search has not been submitted, the registry has libraries, and QA mode is not active.  Set the libraries
+    // variable to a filtered list of libraries--just the ones that are in production.
+    // 5) A search has not been submitted, the registry has libraries, and QA mode is active. Set the libraries variable
+    // to the full list of the registry's libraries.
+    // 6) A search has not been submitted, and the registry does not have libraries. Let the libraries
+    // variable remain undefined; this will generate a "no libraries in this registry" message.
+    return (filteredResults || allResults || noResults) || filteredLibraries || allLibraries;
+  }
+
+  async toggleQA(showQA: boolean) {
+    this.setState({ "qa": showQA });
+    let hasAlreadyLoadedQA: boolean = this.props.libraries.libraries && !(this.props.libraries.libraries.every(l => l.stages.registry_stage === "production"));
+
+    if (!hasAlreadyLoadedQA) {
+      let fetch = showQA ? this.props.fetchQA : this.props.fetchData;
```

Comment:
If you initially called `fetchData` for the production libraries and only want to call it once, do you need this here?

Reply:
Pretty sure I do not; good catch!